src/hotspot/share/prims/foreign_globals.cpp

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2020, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+#include "precompiled.hpp"
+#include "foreign_globals.hpp"
+#include "classfile/symbolTable.hpp"
+#include "memory/resourceArea.hpp"
+#include "runtime/fieldDescriptor.hpp"
+
+#define FOREIGN_ABI "jdk/internal/foreign/abi/"
+
+int ForeignGlobals::field_offset(InstanceKlass* cls, const char* fieldname, Symbol* sigsym) {
+  Symbol* fieldnamesym = SymbolTable::new_symbol(fieldname, (int)strlen(fieldname));
```

Comment:
Use TempNewSymbol fieldname_sym = SymbolTable::new_symbol() here so that the Symbol reference in this scope doesn't leak (destructor of TempNewSymbol refcounts).   Same comment for all of these functions.